### DISCUSSION TITLE
I wanted to carry out Regional simulation and use SOIL_DATA_OPTION = 2, but after I prepared the driver data, the model could not find SOILCL1 after running

### DISCUSSION THREAD

**Discussion by chuxiaoyu001** [NCAR/hrldas#197] [Q&A] [answered]
For Noah - MP of Regional simulation, I downloaded the data on the https://www2.mmm.ucar.edu/wrf/users/download/get_sources_wps_geog.html, I have prepared geo_em.d01.nc data containing soilgrids data (including SOILCL1, SOILCL2, SOILCL3, and SOILCL4). When I prepare the forcing data and carry out simulation, the modle said that SOILCL1 is not found. Do I need to modify the code create_forcing.F about preparing the driver data, or am I doing it wrong? (I'm going to run a simulation of the Tibetan Plateau, using GLDAS data).
Thank you

WPS:
Parsed 67 entries in GEOGRID.TBL
Processing domain 1 of 1
  Processing XLAT and XLONG
  Processing MAPFAC
  Processing F and E
  Processing ROTANG
  Processing LANDUSEF
  Calculating landmask from LANDUSEF ( WATER =  17  21 )
  Processing HGT_M
  Processing SOILTEMP
  Processing SOILCTOP
  Processing SCT_DOM
  Processing SOILCBOT
  Processing SCB_DOM
  Processing ALBEDO12M
  Processing GREENFRAC
  Processing LAI12M
  Processing SNOALB
  Processing CON
  Processing VAR
  Processing OA1
  Processing OA2
  Processing OA3
  Processing OA4
  Processing OL1
  Processing OL2
  Processing OL3
  Processing OL4
  Processing VAR_SSO
  Processing CROPTYPE
  Processing SEASON_GDD
  Processing PLANTING
  Processing HARVEST
  Processing IRFRACT
  Processing SIFRACT
  Processing MIFRACT
  Processing FIFRACT
  Processing TD_FRACTION
  Processing FDEPTH
  Processing RECHCLIM
  Processing EQZWT
  Processing RIVERBED
  Processing SOILCOMP
  Processing SOILCL1F
  Processing SOILCL1
  Processing SOILCL2F
  Processing SOILCL2
  Processing SOILCL3F
  Processing SOILCL3
  Processing SOILCL4F
  Processing SOILCL4

  Optional fields not processed by geogrid:
    LAKE_DEPTH (priority=1, resolution='default', path='/home/cxy/WPS/GEOG/lake_depth/')
    URB_PARAM (priority=1, resolution='default', path='/home/cxy/WPS/GEOG/NUDAPT44_1km/')
    FRC_URB2D (priority=1, resolution='default', path='/home/cxy/WPS/GEOG/urbfrac_nlcd2011/')
    IMPERV (priority=1, resolution='default', path='/home/cxy/WPS/GEOG/nlcd2011_imp_ll_9s/')
    CANFRA (priority=1, resolution='default', path='/home/cxy/WPS/GEOG/nlcd2011_can_ll_9s/')
    EROD (priority=1, resolution='default', path='/home/cxy/WPS/GEOG/erod/')
    CLAYFRAC (priority=1, resolution='default', path='/home/cxy/WPS/GEOG/clayfrac_5m/')
    SANDFRAC (priority=1, resolution='default', path='/home/cxy/WPS/GEOG/sandfrac_5m/')
    IRRIGATION (priority=1, resolution='default', path='/home/cxy/WPS/GEOG/irrigation/fao/')
    CONLS (priority=1, resolution='default', path='/home/cxy/WPS/GEOG/orogwd3_10m/conls/')
    VARLS (priority=1, resolution='default', path='/home/cxy/WPS/GEOG/orogwd3_10m/varls/')
    OA1LS (priority=1, resolution='default', path='/home/cxy/WPS/GEOG/orogwd3_10m/oa1ls/')
    OA2LS (priority=1, resolution='default', path='/home/cxy/WPS/GEOG/orogwd3_10m/oa2ls/')
    OA3LS (priority=1, resolution='default', path='/home/cxy/WPS/GEOG/orogwd3_10m/oa3ls/')
    OA4LS (priority=1, resolution='default', path='/home/cxy/WPS/GEOG/orogwd3_10m/oa4ls/')
    OL1LS (priority=1, resolution='default', path='/home/cxy/WPS/GEOG/orogwd3_10m/ol1ls/')
    OL2LS (priority=1, resolution='default', path='/home/cxy/WPS/GEOG/orogwd3_10m/ol2ls/')
    OL3LS (priority=1, resolution='default', path='/home/cxy/WPS/GEOG/orogwd3_10m/ol3ls/')
    OL4LS (priority=1, resolution='default', path='/home/cxy/WPS/GEOG/orogwd3_10m/ol4ls/')
    CONSS (priority=1, resolution='default', path='/home/cxy/WPS/GEOG/orogwd3_10m/conss/')
    VARSS (priority=1, resolution='default', path='/home/cxy/WPS/GEOG/orogwd3_10m/varss/')
    OA1SS (priority=1, resolution='default', path='/home/cxy/WPS/GEOG/orogwd3_10m/oa1ss/')
    OA2SS (priority=1, resolution='default', path='/home/cxy/WPS/GEOG/orogwd3_10m/oa2ss/')
    OA3SS (priority=1, resolution='default', path='/home/cxy/WPS/GEOG/orogwd3_10m/oa3ss/')
    OA4SS (priority=1, resolution='default', path='/home/cxy/WPS/GEOG/orogwd3_10m/oa4ss/')
    OL1SS (priority=1, resolution='default', path='/home/cxy/WPS/GEOG/orogwd3_10m/ol1ss/')
    OL2SS (priority=1, resolution='default', path='/home/cxy/WPS/GEOG/orogwd3_10m/ol2ss/')
    OL3SS (priority=1, resolution='default', path='/home/cxy/WPS/GEOG/orogwd3_10m/ol3ss/')
    OL4SS (priority=1, resolution='default', path='/home/cxy/WPS/GEOG/orogwd3_10m/ol4ss/')

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
!  Successful completion of geogrid.        !
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

Noah-MP: 
read_soil_composition:  Problem finding variable 'SOILCL1' in NetCDF file.

**Comment by cenlinhe** [comment, marked as answer]
THe SOILCL1~4 should be included in the HRLDAS_setup_file. Please check your HRLDAS_setup file to see if it has these variables. If not, please add these variables from geo_em file directly to hrldas_setup file (e.g., you could use the nco to do it in one single command: https://stackoverflow.com/questions/68635806/how-to-add-variable-from-nc1-to-nc2-file-without-deleting-removing-the-variables)
